src/command/PlayerCommands.cxx

```diff
@@ -142,7 +143,7 @@ handle_status(Client &client, gcc_unused Request args, Response &r)
 		 COMMAND_STATUS_STATE ": %s\n",
 		 playlist.GetRepeat(),
 		 playlist.GetRandom(),
-		 playlist.GetSingle(),
+		 (int)playlist.GetSingle(),
```

Comment:
This `int` cast is suspicious. Are you aware what this will do for the `oneshot` case? Is this what you wanted it to do?
Your patches don't contain protocol documentation updates, but that would have been interesting to read, how you thought the protocol would be extended here.